staging/src/k8s.io/api/resource/v1beta1/types.go

```diff
@@ -171,19 +174,22 @@ type ResourceSliceSpec struct {
 	// SharedCounters defines a list of counter sets, each of which
 	// has a name and a list of counters available.
 	//
-	// The names of the SharedCounters must be unique in the ResourceSlice.
+	// The names of the counter sets must be unique in the ResourcePool.
+	//
+	// Only one of Devices and SharedCounters can be set in a ResourceSlice.
 	//
-	// The maximum number of SharedCounters is 32.
+	// The maximum number of counter sets is 8.
 	//
 	// +optional
 	// +listType=atomic
 	// +featureGate=DRAPartitionableDevices
+	// +zeroOrOneOf=ResourceSliceType
 	SharedCounters []CounterSet `json:"sharedCounters,omitempty" protobuf:"bytes,8,name=sharedCounters"`
```

Comment:
For enforcing "maximum number of counter sets is 8", can use the DV tag:
// +k8s:maxItems=8